libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1375,9 +1394,23 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context.appendConditionalJumpTo(m_context.errorTag());
 
 		if (_operator == Token::Div)
-			m_context << (c_isSigned ? Instruction::SDIV : Instruction::DIV);
+		{
+			if (isFractional)
+			{
+				//divide, then shift left...this is your integer.
+				m_context << (isSigned ? Instruction::SDIV : Instruction::DIV) << fractionShift << Instruction::MUL;
+				//now redo the process...
+				m_context << Instruction::DUP2 << Instruction::DUP4;
+				//but this time, get the fraction portion.
+				m_context << fractionShift << Instruction::SWAP1 << Instruction::MUL << (isSigned ? Instruction::SDIV : Instruction::DIV);
+				//add
+				m_context << Instruction::ADD;
+			}
+ 			else
+				m_context << (isSigned ? Instruction::SDIV : Instruction::DIV);
```

Comment:
This is shared with the `isFractional` case. It could be moved to the top of the block.